src/io/memory.cr

```diff
@@ -225,7 +225,9 @@ class IO::Memory
     if pos == @bytesize
       ""
     else
-      String.new(@buffer + @pos, @bytesize - @pos)
+      String.new(@buffer + @pos, @bytesize - @pos).tap {
```

Comment:
Uh well my personal style certainly isn't, the formatter doesn't seem to enforce it and https://crystal-lang.org/docs/conventions/coding_style.html doesn't mention anything about it.